### ISSUE TITLE
Deleted export /server/index.html, cant re-generate it.

### ISSUE THREAD

**Issue by FrozenBeard** [Rsl1122/Plan-PlayerAnalytics#905]
Ahoy!

I setup some other staff users able to access my analytics and then noticed there was some information I didnt want them to see, so I removed this from the settings file, reloaded the plugin and run /plan analyze (To generate the files?) I have /plan analyze run on a cronjob twice a day to run for my export folder as I have the internal web-server disabled. Is this still the right method?

Either way, I deleted my /server/index.html file and now it wont re-generate as when I removed the setting from the config it didn't update the main web UI.

**Comment by Rsl1122** [comment]
Plan analyze is still the command that should perform the file export.

Here are the settings for export that were used during testing
```
Export:
    HTML_Export_path: 'Analysis Results'
    JSON_Export_path: 'Raw JSON'
    Parts:
        JavaScript_and_CSS: true
        Player_pages: true
        Players_page: true
        Server_page: true
        Player_JSON: false
        Server_JSON: false
```

Do your settings match?

**Comment by FrozenBeard** [comment]
Ok after updating to 4.7.0 it seems to have re-generated the index file!
A few minor issues:

- ~~I enabled buycraft to see the data it generated, but now I removed it, did /plan reload then /plan analyze, but the website still shows the data, is it cached? not updated? ~~ It must have been cache
- I use Export on my own web server, when it exports it gives me the URL I would export sub.domain/server/PirateCraft (the name of the server!) this is shown in-game.
Plan is installed on the paper server, but not on the bungee, but the server is on a bungee network, and the files load from sub.domain/server/ instead of sub.domain/server/PirateCraft

**Comment by Rsl1122** [comment]
Alright, so can this issue be closed?

----

What I understood, the minor issues is:

You access your stuff from `sub.domain/server/` - and export places them in this folder, but /plan analyze command gives address to `sub.domain/server/PirateCraft`, you would like the command to give address to `sub.domain/server/` instead. 

I recommend adding a redirect rule for /server/PirateCraft (This is what Plan webserver does if not on bungee).

The name is added to the address in case the server is running on BungeeCord and multiple servers would be present

**Comment by FrozenBeard** [comment]
Yes the issue was resolved in the main ticket, thank you.
Sorry I should have created a new ticket for my URL questions. 
I will create a re-direct, easy fix.

For future reference, if I was to move to the bungee setup, I install plan on all servers+the bungee right? I assume the bungee instance just generates an index file in the export folder root as there isn't one now? as I cant use sub.domain, I have to add the /server on the end.

Just trying to wrap my head around where/how files are generated!

**Comment by Rsl1122** [comment]
I would not recommend moving to a Bungee system if you want to continue current form of exporting the files due to some limitations

## How export works on Bungee

Plan on Bungee requires (at the moment) both Bungee and Bukkit Web servers to be able to communicate between each other.

- Bukkit server generates a `/server` page -> sends it to Bungee via WebServer connection -> Bungee exports the contents to `<export folder>/server/ServerName/index.html` (if export is enabled)
  - Bukkit server does not export files if it detects that a Bungee information is in the database, even if the Bungee connection fails
- `/player`, `/network` and `/players` pages are generated on Bungee (Although I suspect #958 also applies to network page)
  - `/network` page is placed to `/<export folder>/network/index.html`

I might remember the paths wrong

## Possible workarounds

### Workaround 1

Keeping only exported files accessible to outside would require setting the Web servers to only be accessible in the local network. The commands would give an inaccessible address in this case. (`Alternative_IP` is used for connecting, but `External_Webserver_Address` would be required by the user. The commands give `Alternative_IP` when the webserver is enabled.)

### Workaround 2

Installing the plugin only on all Bukkit servers and exporting player pages on a single server, while exporting the server pages on all servers. This might require moving the server page files around.

## How this will be resolved in the long run

Current long term goal is to eliminate the Bungee-Bukkit Connection system (works via WebServers).

Current limitations that make the system required:
- Analysis takes a lot of memory, so it is performed on each server independently and results are sent to Bungee for serving
- PluginData can only be accessed on the server they are on. This will be resolved by PluginData rework #638

The goal is to get both these resolved. After the connection system will be removed, it will be possible to export all files on a single server.

**Comment by FrozenBeard** [comment]
Fantastic, thank you again for explaining how the current system works. I have a few servers connected via bungee, but my survival is my core playerbase, so I have just stuck to exporting on my bukkit/paper server, hosting the webserver myself gives a lot more freedom to do as I wish, as long as I have an export I can manipulate it. I was thinking about another way to host exports from Plan the other day using netlify/github pages, by just pushing a commit instead of exporting or hosting a web server, since its all static content, commit changes as they happen and netlify would handle the hosting/ssl/backups/history/CDN.
its not something I need, but the thought crossed my mind as a simplified solution for plan hosting, Just a random thought you might like to play with https://www.netlify.com/docs/api/

I will stick to workaround 2 for now! thank you.
